catalog/spatial/registry/registry-api-impl/src/main/java/org/codice/ddf/registry/api/impl/RegistryStoreImpl.java

```diff
@@ -113,16 +128,82 @@ public boolean isPullAllowed() {
     }
 
     @Override
+    public boolean isAutoPush() {
+        return autoPush;
+    }
+
+    @Override
+    public CreateResponse create(CreateRequest request) throws IngestException {
+        Optional nonRegistryMetacard = request.getMetacards()
```

Comment:
Instead of this and the conditional `isPresent()` below, you could just do
```java
if (request.getMetacards()
        .stream()
        .map(this::getRegistryId)
        .anyMatch(Objects::nonNull)) {
    throw new IngestException("One or more of the metacards is not a registry metacard");
}
```
This makes your intent clearer, avoids creating an object you don't need, and tightens up your predicate logic by adding an explicit map call.